### PR TITLE
hw6: redux

### PR DESCRIPTION
**Pull request by kamiour**
hw6: redux
- Implemented Redux
- used `react-toolkit`
- linked with backend API
- All data fetches are implemented as actions and received from store by components
- Sorting by release date and rating is implemented as redux actions
- Filtering by genre is implemented as redux actions

### CODE REVIEW

File: src/store/moviesReducer.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      };
+    },
+    setSelectedMovie: (state: FetchedMoviesState, action: PayloadAction<Movie | null>) => {
+      state.selectedMovie = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(fetchMovies.pending, (state: FetchedMoviesState) => {
+      state.isLoading = true;
+    });
```
Comment (Twixs) [inline src/store/moviesReducer.ts:55]: every time we do request it's better to also clean isError property

File: src/store/moviesApiService.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import axios, { AxiosResponse } from 'axios';
+import { MovieResponse } from '../models/MovieResponse';
+import { QueryParams } from '../models/QueryParams';
+import { initialQueryParams } from './initialQueryParams';
+
+const MOVIES_API_URL = 'http://localhost:4000/movies';
+
+export class MoviesApiService {
+  static async fetchMoviesFromServer(queryParams: QueryParams = initialQueryParams) {
```
Comment (Twixs) [inline src/store/moviesApiService.ts:9]: fetchMoviesFromServer --> fetchMovies ?

File: src/store/initialQueryParams.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { QueryParams } from '../models/QueryParams';
```
Comment (Twixs) [inline src/store/initialQueryParams.ts:1]: either move `initialQueryParams.ts` to /store/utils or inside moviesReducer. In /store folder prefer keeping only reducers, actions, selectors.

File: src/containers/MoviesListOptionsContainer/MoviesListOptionsContainer.tsx [hunk 4]
```diff
@@ -1,39 +1,40 @@
...
-    setQueryParams({ ...currentQueryParams, genre: value });
+  const handleSortBychange = (selectValue: SelectValue): void => {
+    dispatch(setSortBy(selectValue.value));
   };
 
-  const handleSortBychange = (selectValue: SelectValue) => {
-    setSortBy(selectValue);
-    setQueryParams({ ...currentQueryParams, sort: selectValue.value });
+  const getSortByValue = (value: string): SelectValue => {
+    return sortOptions.find((option) => option.value === value)!;
```
Comment (Twixs) [inline src/containers/MoviesListOptionsContainer/MoviesListOptionsContainer.tsx:28]: Why there is ! in the end of line?

File: src/store/moviesReducer.ts (hunk 1 above)
Comment (kamiour) [reply src/store/moviesReducer.ts:55]: Yes, thank you, fixed

File: src/store/moviesApiService.ts (hunk 2 above)
Comment (kamiour) [reply src/store/moviesApiService.ts:9]: Fixed, thanks

File: src/store/initialQueryParams.ts (hunk 3 above)
Comment (kamiour) [reply src/store/initialQueryParams.ts:1]: Thank you, moved to /store/utils

**Review by Twixs — APPROVED** [review]
LGTM

**Pull request merged**